modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/openapi2SpringBoot.mustache

```diff
@@ -8,17 +8,24 @@ import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.context.annotation.FullyQualifiedAnnotationBeanNameGenerator;
 
-@SpringBootApplication
-@ComponentScan(basePackages = {"{{basePackage}}", "{{apiPackage}}" , "{{configPackage}}"})
+@SpringBootApplication(
+    nameGenerator = FullyQualifiedAnnotationBeanNameGenerator.class
```

Comment:
Should this be set for all projects? Or conditionally set?

Reply:
Thanks for taking a look! Is there a downside to having these longer, non-colliding unique bean names as default?

As I understand it, these bean name are rarely if ever used/needed, the beans are autowired via classes / implementations.

At present, there are multiple ways for name collisions to happen (which fail startup), the most common is Spring's `@Bean` annotation defaulting to method name (note the nameGenerators aren't used for those based on my searches, the only way is to specify in the annotation).

And so far I've found the enum converter case to be the most likely of those here. If multiple schemas define a v1 and v2 `FooBar` enum, the same generated method names of `fooBarConverter` will fail startup. Or foreseeably, same problem if down the road entirely separate code happens to use the same method naming pattern.

This would instead default to unique `gen.v1.EnumConverterConfiguration.fooBarConverter` and `gen.v2.EnumConverterConfiguration.fooBarConverter`, and those bean names are never used.

I could remove all the `nameGenerator`s changes to reduce the blast radius of this PR, but I can't get past this particular enum class without explicitly specifying a unique name for `@Bean` here, and it's my hope that this could be built in to the generator rather than another config option.

Reply:
@welshm 
The syntactically incorrect generated java is obviously the most pressing error, I made a separate PR for now, would love to get it merged faster:
https://github.com/OpenAPITools/openapi-generator/pull/14104

In the meantime I can hack around default `@Bean` name collisions with adding an exclude filter to my root `@SpringApplication`, and using reflection to manually register these converters with unique names.

Reply:
I think the argument you've presented makes sense, in that most of the time the name shouldn't be necessary if autowiring is done, so using the full class name shouldn't be a problem as a default.

I'm on board with this change since I don't think this will have any visible impact on downstream users of the generator (they'll get more specific bean names, but shouldn't have any issues if they were already using autowiring).